lib/db/set.go

```diff
@@ -146,29 +152,39 @@ func NewFileSet(folder string, fs fs.Filesystem, db *Instance) *FileSet {
 
 func (s *FileSet) Replace(device protocol.DeviceID, fs []protocol.FileInfo) {
```

Comment:
Given that it does now something else than before and is only used once, I would personally remove it entirely and call `Drop` and `Update` explicitly the one time it is used in the model.